### DISCUSSION TITLE
Add metadata as a watermark

### DISCUSSION THREAD

**Discussion by acorderob** [vladmandic/sdnext#2045] [Ideas]
I just had this idea, and i think it could be really useful. It would mean appending a formatted parameters text to the watermark text selected by the user, or, probably better, had the user text just be inside it.

Pros:

* Shared images might keep the metadata even if the provider strips regular metadata.
* It would allow us to define a more parseable metadata format (JSON?) and maybe start escaping from the mess that is the current one.

Cons:

* I tried regenerating an image with its metadata as a watermark, and the result was a very noticeable pixelated color shifting. It's the first time i use this function, and this might be related to the algorithm used or its configuration, but right now this watermark seems unusable. Watermarks are supposed to be invisible.
* It's less user readable since it needs specific software. But it could also be added to the regular metadata in a way that would not interfere with the current format.

**Comment by vladmandic** [comment]
Custom watermark is already supported in SDnext.
But watermarks are limited to short strings - encoding long string (json or not) is not feasible without massive image degradation.

**Reply by acorderob** [reply]
My test seemed to work (although i didn't read the watermark afterwards), and it degraded the image almost the same as with a short string (both unacceptable), I guess it all depends on the algorithm.

**Comment by vladmandic** [comment]
> My test seemed to work (although i didn't read the watermark afterwards)

sorry, but that's not really a test. i'm pretty sure that string got cropped to short value, no chance you can embed a long string in 512x512 image at all - in my test anything above few chars starts getting corrupted really fast.
